### PR TITLE
Add arguments to pyfmt so that you may skip files and directories

### PR DESCRIPTION
**Pull request by ismet55555**
Added a `--skip` option for pyfmt to skip files and/or directories
Useful for when you have to skip the virtual environment (`env`, `venv`, etc)  directory or something like that.

Example for skipping directories. Will skip everything in the directory.
It must be relative to current directory format. 
- `--skip=dir/subdir`

Example for specific files. Will skip specific files.
- `--skip=yo.py,mama.py`

Example for mixing things up with files and directories:
- `--skip=daddy.py,dir/subdir`

If it can't find a file or directory it will stop pyfmt with a error message. It would be pretty bad to want to skip a file, misspell it, and it formats it anyways!

Basically:
- isort: --skip
- black: --exclude

**don't judge my crapy programming**

### CODE REVIEW

File: pyfmt/__init__.py [hunk 1]
```diff
@@ -25,8 +27,121 @@
 ]
 
 
-def pyfmt(path, check=False, line_length=100, extra_isort_args="", extra_black_args="") -> int:
+def pyfmt_title():
+    """
+    Display a cool ascii art title
```
Comment (rtrajano) [inline pyfmt/__init__.py:33]: ✨

File: pyfmt/__init__.py [hunk 2]
```diff
@@ -25,8 +27,121 @@
...
+                # Saving all filenames in directory
+                files_in_dir = list()
+                for (dirpath, dirnames, filenames) in os.walk(item):
+                    for filename in filenames:
+                        if filename.split(".")[-1] == "py" or filename.split(".")[-1] == "pyi":
+                            files_in_dir.append(filename)
+                filenames_to_skip.extend(files_in_dir)
+            else:
+                print(
+                    'CRITICAL: One of the files or directories marked as skipped not found ("{}").'.format(
```
Comment (rtrajano) [inline pyfmt/__init__.py]: maybe u could add a logger to the file and use logger.error for these messages and for the timing stuff, logger.info

Comment (ismet55555) [reply pyfmt/__init__.py]: Yeah I really thought about it, but then I thought since this is a one-off kind of deal and there are now only like a couple of messages that may never even pop up, it may be overkill.  Plus it has all the timestamp and other features that just seem overwhelming for this application.

Then you say, "you can remove the timestamps and all", then why have the logger?  for log aggregation for later?

Maybe i'm thinking about it all wrong

File: pyfmt/__init__.py [hunk 3]
```diff
@@ -25,8 +27,121 @@
...
+        for filename_to_skip in filenames_to_skip:
+            print(f"SKIPPING: {filename_to_skip}")
+        print("\nNumber of files to be skipped: {}".format(len(filenames_to_skip)))
+        # Make a continuos string of arguments for
+        #   isort - must be separate --skip for each file
+        #   black - regex for exact filename (ie. file1|file2|etc.)
+        isort_filenames_to_skip = ""
+        black_filenames_to_skip = ""
+        for filename in filenames_to_skip:
+            isort_filenames_to_skip += "--skip=" + filename + " "
```
Comment (augustawind) [inline pyfmt/__init__.py]:
```suggestion
            isort_filenames_to_skip += f"--skip={filename} "
```

File: pyfmt/__init__.py [hunk 4]
```diff
@@ -25,8 +27,121 @@
...
+            black_filenames_to_skip += filename + "|"
```
Comment (augustawind) [inline pyfmt/__init__.py]:
```suggestion
            black_filenames_to_skip += f"{filename}|"
```

File: pyfmt/__init__.py [hunk 5]
```diff
@@ -25,8 +27,121 @@
...
+        isort_filenames_to_skip = isort_filenames_to_skip[:-1]
+        black_filenames_to_skip = black_filenames_to_skip[:-1]
```
Comment (augustawind) [inline pyfmt/__init__.py]: why this?

File: pyfmt/__init__.py [hunk 6]
```diff
@@ -25,8 +27,121 @@
...
+                    )
+                )
+                print("CRITICAL: Check spelling or existence of file or directory")
+                print("CRITICAL: Aborting pyfmt ...")
+                sys.exit()
+        # Display Files skipped
+        display_divider(title="SKIPPING FILES")
+        for filename_to_skip in filenames_to_skip:
+            print(f"SKIPPING: {filename_to_skip}")
+        print("\nNumber of files to be skipped: {}".format(len(filenames_to_skip)))
```
Comment (augustawind) [inline pyfmt/__init__.py]:
```suggestion
        print(f"\nNumber of files to be skipped: {len(filenames_to_skip)}")
```

File: pyfmt/__init__.py (hunk 2 above)
Comment (augustawind) [inline pyfmt/__init__.py]:
```suggestion
                    f'CRITICAL: One of the files or directories marked as skipped not found ("{item}").'
```

File: pyfmt/__init__.py [hunk 7]
```diff
@@ -25,8 +27,121 @@
...
+        for item in skips:
+            if item in all_files:
+                if item.split(".")[-1] == "py":
+                    filenames_to_skip.extend([item])
+            elif os.path.abspath(item) in all_dirs:
+                # Saving all filenames in directory
+                files_in_dir = list()
+                for (dirpath, dirnames, filenames) in os.walk(item):
+                    for filename in filenames:
+                        if filename.split(".")[-1] == "py" or filename.split(".")[-1] == "pyi":
```
Comment (augustawind) [inline pyfmt/__init__.py]:
```suggestion
                        if filename.split(".")[-1] in ("py", "pyi"):
```

File: pyfmt/__init__.py [hunk 8]
```diff
@@ -25,8 +27,121 @@
...
+        # If specified files and directories exist, store the filenames
+        skips = skip.split(",")
+        filenames_to_skip = []
+        for item in skips:
+            if item in all_files:
+                if item.split(".")[-1] == "py":
+                    filenames_to_skip.extend([item])
+            elif os.path.abspath(item) in all_dirs:
+                # Saving all filenames in directory
+                files_in_dir = list()
```
Comment (augustawind) [inline pyfmt/__init__.py]:
```suggestion
                files_in_dir = []
```

using `[]` is slightly more efficient than `list()` 🙂

File: pyfmt/__init__.py [hunk 9]
```diff
@@ -25,8 +27,121 @@
...
+            + "".center(os.get_terminal_size().columns, "{}".format(character))
+            + "\033[0m"
+        )
+
+
+def pyfmt(
+    path, skip="", check=False, line_length=100, extra_isort_args="", extra_black_args=""
+) -> int:
     """Run isort and black with the given params and print the results."""
+    pyfmt_title()  # Display title
```
Comment (augustawind) [inline pyfmt/__init__.py]: while I like the ascii art you made, but I'm not suree I want to see a title everytime pyfmt runs. it will fill up my terminal with spam since I run it so often

File: pyfmt/__init__.py [hunk 10]
```diff
@@ -47,11 +163,12 @@ def run_formatter(cmd, path, **kwargs) -> int:
     result = subprocess.run(cmd, stdout=PIPE, stderr=PIPE)
 
     prefix = f"{cmd[0]}: "
-    sep = "\n" + (" " * len(prefix))
+    display_divider(title=prefix[:-2])
+    sep = "\n"
     lines = result.stdout.decode().splitlines() + result.stderr.decode().splitlines()
     if "".join(lines) == "":
-        print(f"{prefix}No changes.")
```
Comment (augustawind) [inline pyfmt/__init__.py:53]: why remove the prefixes?

File: pyfmt/__init__.py (hunk 5 above)
Comment (ismet55555) [reply pyfmt/__init__.py]: remove that last character, " " and "|" in each string ... probably a better way to do it

File: pyfmt/__init__.py (hunk 7 above)
Comment (ismet55555) [reply pyfmt/__init__.py]: yeah i actually thought about this one while i was doing it,... good change though

File: pyfmt/__init__.py (hunk 10 above)
Comment (ismet55555) [reply pyfmt/__init__.py:53]: I added a divider of each module's work, didn't think the indentation is necessary
...BUT this is completely subjective!  I'm ok with either way

File: pyfmt/__init__.py (hunk 9 above)
Comment (ismet55555) [reply pyfmt/__init__.py]: i don't know seemed neat when i was doing it haha ... again, subjective, i'm good with leaving it out or replacing it with something more compact

File: pyfmt/__init__.py [hunk 11]
```diff
@@ -108,15 +108,15 @@ def pyfmt(
...
-                        if filename.split(".")[-1] == "py" or filename.split(".")[-1] == "pyi":
+                        if filename.split(".")[-1] in ("py", "pyi"):
                             files_in_dir.append(filename)
                 filenames_to_skip.extend(files_in_dir)
             else:
                 print(
-                    'CRITICAL: One of the files or directories marked as skipped not found ("{}").'.format(
+                    f'CRITICAL: One of the files or directories marked as skipped not found ("{item}").'
                         item
                     )
```
Comment (augustawind) [inline pyfmt/__init__.py]: this is a syntax error

File: pyfmt/__init__.py (hunk 5 above)
Comment (augustawind) [reply pyfmt/__init__.py]: ah yeah, you could do this instead:

```python
isort_filenames_to_skip = "--skip=" + " --skip=".join(filenames_to_skip)
black_filenames_to_skip = "--exclude=" + "|".join(filesnames_to_skip)
```

instead of the `for` loop above

File: pyfmt/__init__.py (hunk 2 above)
Comment (augustawind) [reply pyfmt/__init__.py]: yeah I'd say that's overkill imo, this is just a dev tool. doesn't need to be fancy.

File: pyfmt/__init__.py (hunk 9 above)
Comment (augustawind) [reply pyfmt/__init__.py]: at the very least if you could make it an opt-in sort of thing, like if the user passes `--show-title` or something it shows but otherwise it doesn't

File: pyfmt/__init__.py (hunk 10 above)
Comment (augustawind) [reply pyfmt/__init__.py:53]: the prefix isn't indentation, it's just `isort: ` or `black: ` so you know which tool the output is from

Comment (ismet55555) [reply pyfmt/__init__.py]: no i get that, prefix is still there, but in the divider now, but i removed the prefix as in line with the indentation like it was

File: pyfmt/__init__.py (hunk 9 above)
Comment (ismet55555) [reply pyfmt/__init__.py]: yeah i can do that

File: pyfmt/__init__.py (hunk 10 above)
Comment (ismet55555) [reply pyfmt/__init__.py:53]: no i get that, the prefix is still there but moved to the divider.  I removed the indentation part and the part where the prfix is in line with the indent

File: pyfmt/__init__.py (hunk 11 above)
Comment (ismet55555) [reply pyfmt/__init__.py]: what do you mean?   Not sure, maybe one of those recent changes accepted above?  I just ran it as it was, no syntax error

File: pyfmt/__init__.py (hunk 9 above)
Comment (ismet55555) [reply pyfmt/__init__.py]: yeah i can do that

File: pyfmt/__init__.py (hunk 5 above)
Comment (ismet55555) [reply pyfmt/__init__.py]: much better!

File: pyfmt/__init__.py (hunk 11 above)
Comment (augustawind) [reply pyfmt/__init__.py]: oh nvm this is outdated

**Review by augustawind — DISMISSED** [review]

**Comment by augustawind** [comment]
thanks for the improvements!

File: pyfmt/__init__.py [hunk 12]
```diff
@@ -1,5 +1,6 @@
 import os
 import shlex
+from shutil import get_terminal_size
```
Comment (augustawind) [inline pyfmt/__init__.py:3]: looks like you need to run pyfmt to sort these imports 😛

Comment (ismet55555) [reply pyfmt/__init__.py:3]: haha how ironic

Comment (ismet55555) [reply pyfmt/__init__.py:3]: sec... i'll push an update

Comment (ismet55555) [reply pyfmt/__init__.py:3]: hey never mind, this was right!

Comment (augustawind) [reply pyfmt/__init__.py:3]: i thought `from` imports should be below `import` statements.. weird!

**Review by augustawind — APPROVED** [review]

**Comment by ismet55555** [comment]
Dupplicate from subsequent PR #7
